### DISCUSSION TITLE
Seeking assistance regarding issues related to dry powder fire extinguishing

### DISCUSSION THREAD

**Discussion by MaoyiHe** [firemodels/fds#13767] [Q&A]
I want to conduct a study on dry powder fire extinguishing in tunnels. In Document 1, I only have the working conditions of the dry powder that has extinguishing effects, but around 200 seconds of runtime, I encounter the error 'MPI exchange of particle sizes timed out for MPI process 5 running.' In Document 2, I have introduced thermal decomposition reactions for the dry powder in the hope that it will not only have extinguishing effects but also participate in chemical reactions to extinguish the fire. However, I am facing the error 'minimum density 0.120 kg/m3 clipped in mesh 5.' I sincerely hope that an expert can assist me.
[document1.txt](https://github.com/user-attachments/files/17790741/document1.txt)
[document2.txt](https://github.com/user-attachments/files/17790742/document2.txt)

**Comment by drjfloyd** [comment]
Make sure you are using the current release of FDS

&MATL ID='na2co3',
      SPECIFIC_HEAT=2.9,
      CONDUCTIVITY=0.38,
      DENSITY=2532.0,
      N_REACTIONS=1,
      HEAT_OF_REACTION=50.0,
      SPEC_ID(1,1)='WATER VAPOR',
      SPEC_ID(2,1)='NH3',
      NU_SPEC(1,1)=0.5,
      NU_SPEC(2,1)=0.33,
      REFERENCE_TEMPERATURE=100.0/

How does Na2CO3 decomposie into water vapor and ammonia? There is no nitrogen or hydrogen in sodium carbonate and you are not doing anything with the carbon. Note dry chem extinguishers are generally sodium bicarbonate not sodium carbonate.

&SURF ID='na2co3',
      RGB=146,202,166,
      MATL_ID(1,1)='na2co3',
      MATL_MASS_FRACTION(1,1)=1.0,
      THICKNESS(1)=0.01,
      GEOMETRY='SPHERICAL',
      INNER_RADIUS=0.0/

Your extinguisher discharges 2 cm diameter particles?

You have no inputs for how decomposition products would chemically effect the fire. 

I suggest you start with much simpler inputs. Make a simple case where you just work on developing inputs for the decomposiiton of poweder. A simple case where you develop inputs for how those products of decomposiiton effect the fire. Then slowly buildup to the full simulation. This is very much a research application of FDS.

**Reply by MaoyiHe** [reply]
Thank you for your response. 

First, I made a mistake by writing Na2CO2 instead of NH4H2PO4. My intention was to discuss NH4H2PO4 as a dry powder particle. Second, I defined the heat absorbed by the particle during combustion as 50 kJ/kg, and the decomposition of the dry powder will produce water vapor. Therefore, I would like to explore the impact of this reaction on fire extinguishing. Third, I will continue to modify the diameter of the dry powder. Finally, I would like to ask you a question: does the FDS database contain dry powder particles, such as sodium bicarbonate, that can be directly called?

**Reply by drjfloyd** [reply]
You will need to define all the properties for the MATL.
